node/src/components/small_network.rs

```diff
@@ -1028,22 +970,17 @@ where
                 effects
             }
             Event::LinearChainAnnouncement(LinearChainAnnouncement::BlockAdded(block)) => {
-                // On switch blocks, we need to update our validator sets.
-                if block.header().is_switch_block() {
-                    let era_id = block.header().era_id();
-
-                    if era_id > self.highest_era_seen {
-                        self.highest_era_seen = era_id;
-                        self.handle_active_era_change(effect_builder, era_id)
-                    } else {
-                        debug!(highest_era_seen=%self.highest_era_seen, %era_id,
-                               "ignoring era, as it is not the highest seen");
-                        Effects::new()
-                    }
-                } else {
-                    trace!("ignoring non-switch block");
-                    Effects::new()
-                }
+                if let Some(next_era_validators) =
+                    block.take_header().maybe_take_next_era_validator_weights()
+                {
+                    let upcoming_validators: HashSet<PublicKey> =
+                        next_era_validators.into_keys().collect();
+                    self.outgoing_limiter
+                        .update_validators(Default::default(), upcoming_validators.clone());
```

Comment:
Shouldn't we treat these as _active_ validators now instead of upcoming? After all, that era is starting right now.

```suggestion
                        .update_validators(upcoming_validators.clone(), Default::default());
```

Reply:
As discussed, @marc-casperlabs plans to take care of this in an upcoming PR.

Reply:
Still, wouldn't it be better to use the first argument here instead of the second? (Not sure what difference it makes…?)